.github/workflows/codespell.yml

```diff
@@ -0,0 +1,18 @@
+on:
+    push:
+        branches: [master]
+    pull_request:
+
+name: Spell Check
+ 
+jobs:
+    codespell:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v2
+            - uses: codespell-project/actions-codespell@master
+              with:
+                  check_filenames: true
+                  only_warn: 1
+                  ignore_words_list: ba,probleme,formend,compiletime,chage,hve,files,ans,inout,testin,files\',creat
+                  skip: .idea,.git,*.guess,*.png,*.zip,*.log,*.sub,Makefile*,ChangeLog,uploadcodecov.sh,codespell.yml,tmp,lib,var,public
```

Comment:
Why skip especially ChangeLog, but also Makefiles? Also, does this match files only at the root of the project or also deeper down? If the latter, then skipping `lib,var,public` may have unintented consequences.

Reply:
We skip ChangeLog because it contains some typos but I think in the past we agreed to not change the ChangeLog file after usage of such tools.

This will also do deeper down. I think the only var we have is webapp/var which we can change to cache as that will contain upstream code which we dont want to test.

public should target webapp/public`
but will target:
./lib/vendor/zircote/swagger-php/docs/.vuepress/public
./lib/vendor/nelmio/api-doc-bundle/Resources/public
./webapp/templates/public
./webapp/public

the lib/vector is fine. the webapp/templates/public would be nice to target.

lib will target:
./lib
./lib/vendor/doctrine/data-fixtures/lib
./lib/vendor/doctrine/deprecations/lib
./lib/vendor/doctrine/orm/lib
./lib/vendor/doctrine/inflector/lib
./lib/vendor/doctrine/lexer/lib
./lib/vendor/doctrine/cache/lib
./lib/vendor/doctrine/annotations/lib
./lib/vendor/doctrine/persistence/lib
./lib/vendor/doctrine/common/lib
./lib/vendor/doctrine/migrations/lib
./lib/vendor/doctrine/collections/lib
./lib/vendor/doctrine/dbal/lib
./lib/vendor/doctrine/event-manager/lib
./lib/vendor/nikic/php-parser/lib
./lib/vendor/ircmaxell/password-compat/lib
./lib/vendor/novus/nvd3/examples/lib
./lib/vendor/mbostock/d3/lib 

the lib/vendor is intented, but now I see that there is code there which could be targeted will make a change for it.

var will currently only target:
./webapp/var

Reply:
I don't think we want to include `lib/vendor` as it contains upstream code.

Reply:
what I tried to say was:
lib/vendor was intended to be skipped, the ./lib should not as it has some of our own code so I changed it to vendor which will still trigger ./lib/vendor but we will scan ./lib for spelling mistakes.